File: PAT (Basic-Level)/1027.c
#include <stdio.h>

int main()
{
	int n, remain;
	char ch;
	scanf("%d %c", &n, &ch);
	int i = 1;
	while((2*i*i - 1) <= n){
		i++;
	}
	i--;

	remain = n - 2*i*i + 1;
	int j = 2 * i - 1 ;
	int max = j;
	while(j > 0){
		int tem = i;
		for(tem; tem < (max + 1)/2; tem++)
			printf(" ");
		for(j; j > 0; j--)
			printf("%c", ch);
		printf("\n");
		i--;
		j = 2 * i - 1;
	}
	i += 2;
	j = 2 * i - 1;
	while(j <= max){
		int tem = i;
		for(tem; tem < (max + 1)/2; tem++)
			printf(" ");
		for(j; j > 0; j--)
			printf("%c", ch);
		printf("\n");	
		i++;
		j = 2 * i - 1;
	}
	printf("%d", remain);
	return 0;	
}

File: PAT (Basic-Level)/Python/1027.py
#!/usr/bin/python3
# -*- coding: utf-8 -*-

data = input().split()
i = 1
while 2*i*i-1 <= int(data[0]):
    i += 1
i -= 1
remain = int(data[0])-2*i*i+1
j = 2*i-1
max = j

while j > 0:
    tem = i
    for k in range(tem, int((max+1)/2)):
        print(" ", end = '')
    for k in range(j, 0, -1):
        print(data[1], end = '')
    print("\n", end = '')
    i -= 1
    j = 2*i-1

i += 2
j = 2*i-1
while j <= max:
    tem = i
    for k in range(tem, int((max+1)/2)):
        print(" ", end = '')
    for k in range(j, 0, -1):
        print(data[1], end = '')
    print("\n", end = '')
    i += 1
    j = 2*i-1

print(remain, end = '')
